SmartDeviceLink/SDLPermissionManager.m

```diff
@@ -26,6 +26,7 @@ @interface SDLPermissionManager ()
 @property (strong, nonatomic) NSMutableDictionary<SDLPermissionRPCName, SDLPermissionItem *> *permissions;
 @property (strong, nonatomic) NSMutableArray<SDLPermissionFilter *> *filters;
 @property (copy, nonatomic, nullable) SDLHMILevel currentHMILevel;
+@property (assign, nonatomic) BOOL requiresEncryption;
```

Comment:
The logic for setting `requiresEncryption` should be the same as in the `SDLEncryptionLifecycleManager`. Perhaps you should create a utility class that handles the logic for both classes since they are exactly the same.